main: calling foo: 0
    foo: calling baz
    foo: got exception U0; handled and dandy
    foo: finished
main: calling foo: 1
    foo: calling baz
main: Someone threw U1; handled and dandy
main: calling foo: 2
    foo: calling baz
Unhandled exception U2
